test/parallel/test-quic-ipv6only.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+// TODO support ipv6
+const common = require('../common');
+if (!common.hasQuic)
+  common.skip('missing quic');
+
+const { createSocket } = require('quic');
+const fixtures = require('../common/fixtures');
+const key = fixtures.readKey('agent1-key.pem', 'binary');
+const cert = fixtures.readKey('agent1-cert.pem', 'binary');
+const ca = fixtures.readKey('ca1-cert.pem', 'binary');
+
+const kServerName = 'agent2';
+const kALPN = 'zzz';
+
+// Setting `type` to `udp4` while setting `ipv6Only` to `true` is possible
+// and it will throw an error.
+{
+  const server = createSocket({
+    type: 'udp4',
+    port: 0,
+    ipv6Only: true,
+  });
+
+  server.on('error', common.mustCall((err) => {
+    common.expectsError({
+      code: 'EINVAL',
+      type: Error,
+      // TODO(@oyyd): Currently we cann't know the exact "syscall" so that it's
+      // undefined here.
+      message: 'undefined EINVAL',
```

Comment:
Just as a heads up, with https://github.com/nodejs/quic/pull/165 the syscall is correctly set to `bind`.